solutions/cf-1528A.mdx

```diff
@@ -133,4 +133,72 @@ int main() {
 }
 ```
 </CPPSection>
+
+<JavaSection>
+```java
+import java.io.*;
+import java.util.*;
+
+public class ParsasHumongousTree {
+	private static Kattio io = new Kattio();
+	private static List<List<Integer>> adj = new ArrayList<>();
+	private static int[] l, r;
+	private static long[][] dp;
+
+	public static void main(String[] args) {
+		int t = io.nextInt();
+
+		while (t-- != 0) {
+			int n = io.nextInt();
+
+			l = new int[n + 1];
+			r = new int[n + 1];
+			dp = new long[n + 1][2];
+			adj.clear();
+
+			for (int i = 1; i <= n; i++) {
+				l[i] = io.nextInt();
+				r[i] = io.nextInt();
+			}
+
+			for (int i = 0; i <= n; i++) {
+				adj.add(new ArrayList<>());
+			}
+
+			for (int i = 0; i < n - 1; i++) {
+				int u = io.nextInt(), v = io.nextInt();
+				adj.get(u).add(v);
+				adj.get(v).add(u);
+			}
+
+			dfs(1, -1);
```

Comment:
dfs says nothing about the function name

Reply:
i mean the function is just a dfs.. its also what the c++ code uses

Reply:
what does it do during the dfs though?